pkg/azuredisk/nodeserver.go

```diff
@@ -81,17 +81,33 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 	}
 	defer d.volumeLocks.Release(diskURI)
 
+	lun, ok := req.PublishContext[LUN]
+	if !ok {
+		return nil, status.Error(codes.InvalidArgument, "lun not provided")
+	}
+
+	source, err := d.getDevicePathWithLUN(lun)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to find disk on lun %s. %v", lun, err)
+	}
+
+	// If perf optimizations are enabled
+	// tweak device settings to enhance performance
+	if d.getPerfOptimizationEnabled() {
+		err = d.getDeviceHelper().OptimizeDevicePerformance(d.getNodeInfo(), d.getDiskSkuInfoMap(), source, req.GetVolumeContext())
+		// In the beginning, do not fail because of perf optimization related error
+		// TODO: revisit this decision after getting some test runs
+		if err != nil {
+			klog.Errorf("NodeStageVolume: Failed to optimize device performance for target(%s) error(%s)", source, err)
```

Comment:
it's better return error directly, silent warning could be ignored quite easily.